game.rb

```diff
@@ -0,0 +1,128 @@
+require_relative 'dictionary'
+
+class Game
+  attr_reader :word, :guess
+
+  def initialize(word = nil)
+    @dictionary = Dictionary.new.trim_dictionary
+    @word = word || @dictionary.sample(1)[0].split('')
+    @used_letters = []
+    @unused_letters = ('a'..'z').to_a
+    @guess = []
+    @word.each do |_letter|
+      @guess.push('_')
+    end
+    @hangman_counter = 0
+    @hangman_pics = ["
+
+
+
+
+
+=========",
+                     " +---+
+  |   |
+      |
+      |
+      |
+      |
+=========",
+                     "  +---+
+  |   |
+  O   |
+      |
+      |
+      |
+=========",
+                     "  +---+
+  |   |
+  O   |
+  |   |
+      |
+      |
+=========",
+                     "  +---+
+  |   |
+  O   |
+ /|   |
+      |
+      |
+=========",
+                     %q(  +---+
+  |   |
+  O   |
+ /|\  |
+      |
+      |
+=========),
+                     %q(  +---+
+  |   |
+  O   |
+ /|\  |
+ /    |
+      |
+=========),
+                     %q(  +---+
+  |   |
+  O   |
+ /|\  |
+ / \  |
+      |
+=========)]
+  end
+
+  def take_input
+    puts 'Please choose a letter'
+    choice = gets.chomp.downcase
+    until valid?(choice)
+      puts 'Please choose a valid letter'
+      choice = gets.chomp.downcase
+    end
+    choice
+  end
+
+  def valid?(choice)
+    if choice.length != 1
+      puts 'Please enter only one letter'
+      false
+    elsif @used_letters.include?(choice)
+      puts "You've already used this letter"
+      false
+    elsif !@unused_letters.include?(choice)
+      puts 'Please use a valid letter, you can choose from ones below'
+      puts @unused_letters.join(', ')
+      false
+    else
+      true
+    end
+  end
+
+  def lose?
+    @hangman_counter == 7
```

Comment:
Why 7 specifically? What if I a very, very long word gets selected :thinking: ?

Reply:
Ok, nevermind. I see you only increment on incorrect guessess :+1:

Reply:
Because there are seven stages to hangman drawing, I guess it could also be `@hangman_pics.length - 1` or `@hangman_pics.length` if I started it at 1